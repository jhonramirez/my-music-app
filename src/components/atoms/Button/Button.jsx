import "./Button.scss";

export const Button = ({ label, icon }) => {
  return (
    <button className="btn btn__gray">
      {icon}
      {label}
    </button>
  );
};
