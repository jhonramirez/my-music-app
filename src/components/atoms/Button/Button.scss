.btn {
  border-radius: 10px;
  font-size: 1rem;
  padding: 8px;
  border: none;
  cursor: pointer;
  &__black {
    background: rgb(17, 17, 16);
    color: #dad4ce;
  }
  &:hover {
    background: rgb(31, 31, 29);
    border-color: #565e64;
    color: #ffffff;
  }
}
