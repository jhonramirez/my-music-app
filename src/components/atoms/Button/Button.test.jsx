import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./Button";
import "@testing-library/jest-dom";

describe("Pruebas en <Button />", () => {
  const { container } = render(<Button label={"proof"} />);
  const ButtonElement = screen.getByText("proof");

  test("Render Button y match con el snapshot", () => {
    expect(ButtonElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("El className del Button debe ser btn btn__black", () => {
    expect(ButtonElement).toHaveClass("btn btn__black");
  });

  test("Se evalua que se ejecute el onClick", () => {
    const onClick = jest.fn();
    render(<Button label={"proof"} onClick={onClick} />);
    const ButtonElement = screen.getByTestId("button");
    fireEvent.click(ButtonElement);
    expect(onClick).toHaveBeenCalled();
  });
});
