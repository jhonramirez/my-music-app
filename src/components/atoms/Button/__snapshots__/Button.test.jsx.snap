// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Button /> Render Button y match con el snapshot 1`] = `
<div>
  <button
    class="btn btn__black"
    data-testid="button"
  >
    proof
  </button>
</div>
`;
